packages/function-entry/src/index.js

```diff
@@ -0,0 +1,60 @@
+import { ModelSetEvent, SessionChangedEvent } from '@pie-framework/pie-player-events';
+import ReactFunctionEntry from './function-entry.jsx';
+import debug from 'debug';
+import React from 'react';
+import ReactDOM from 'react-dom';
+
+const log = debug('pie-elements:text-entry');
+
+export default class FunctionEntry extends HTMLElement {
+
+  constructor() {
+    super();
+  }
+
+  set model(m) {
+    this._model = m;
+    this.dispatchEvent(new ModelSetEvent(
+        this.tagName.toLowerCase(),
+        false,
+        !!this._model
+    ));
+
+    this.render();
+  }
+
+  set session(s) {
+    this._session = s;
+    this.render();
+  }
+
+  onValueChanged(value) {
+    this._session.value = value;
+
+    log('[onSessionChanged] session: ', this._session);
+
+    this.dispatchEvent(new SessionChangedEvent(
+        this.tagName.toLowerCase(),
+        false,
```

Comment:
I think you should set `complete` here - for now, if the session.value is not empty we can consider it complete?

Reply:
I'm not entirely certain what `complete` represents here. I simply used how I saw elsewhere. I can transmit `true` if that's what we need here.

Reply:
it's up to each element to define what 'complete' means.. if the student has entered a response that typically means that it's `complete` - maybe `answered` or `hasResponse` is a more accurate name for it.